src/public/index.php

```diff
@@ -40,16 +40,16 @@ function handle_exception($e)
 include '../database.php';
 $ji_db = new PDO(
     'mysql:host=' . $db['default']['hostname'] .
-    ';dbname=' . $db['default']['database'],
+    ';dbname=' . $db['default']['database'] . ';charset=utf8mb4',
     $db['default']['username'],
     $db['default']['password']
 );
 $ji_db->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
 
 // Set the correct charset for this connection
 $ji_db->query("SET NAMES 'utf8' COLLATE 'utf8_general_ci'");
```

Comment:
All three of those queries should be unnecessary. Setting the charset during PDO connection time in line 43 is sufficient.